includes/class-utils.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Automattic\LegacyRedirector;
+
+final class Utils {
```

Comment:
Instead of a Utils class, how about a `Url` class, that could be instantiated (avoiding static methods), and then have methods for `normalize` (can defer to a constructor dependency injection of a `Normalizer` interface), `hasher` (can defer to a constructor dependency injection of a `Hasher` interface), `get_preservable_querystring_params()`, and then this `->mb_parse_url( $components )` method?

(Might be easier to go with Utils for now, but then refactor again)

Reply:
I agree here, I have a feeling we will have much more confidence with refactoring as we add more unit/integration tests, it would be a good test against the validity of the tests as well :)